include/dca/phys/dca_step/cluster_solver/stdthread_qmci/stdthread_qmci_cluster_solver.hpp

```diff
@@ -160,10 +160,22 @@ void StdThreadQmciClusterSolver<QmciSolver>::integrate() {
   if (concurrency_.id() == concurrency_.first())
     thread_task_handler_.print();
 
-  std::vector<std::future<void>> futures;
+  std::vector<dca::parallel::thread_traits::future_type<void>> futures;
 
   dca::profiling::WallTime start_time;
 
+#ifdef DCA_HAVE_HPX
```

Comment:
Is it possible to not have #ifdef in high-level code? Can't we use the hpx thread pool to return a hpx::future, without having to write different specializations of this method?

Reply:
I removed the #ifdef things out and thanks for pointing out. Please test it again, thanks!